float64     amplitude
float64     position_offset
float64     period
float64     time_offset 